File: Data Structures & Algorithms/number-of-good-pairs/submission-1.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        res = 0
        m = {}

        for num in nums:
            if num not in m:
                m[num] = 1
            else:
                res += m[num]
                m[num] += 1

        return res
        

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;

        int[] difference = new int[n];
        for (int j = 0; j < n; j++) {
            difference[j] = target - nums[j];
        }

        Map<Integer, Integer> differenceIndex = new HashMap<>();
        for (int i = 0; i < n; i++) {
            differenceIndex.put(difference[i], i);
        }
        Integer j;
        for (int i = 0; i < n; i++) {
            j = differenceIndex.get(nums[i]);
            if (j != null && j > i) {
                int[] result = new int[2];
                result[0] = i;
                result[1] = j;
                return result;
            }
        }
        return null;
    }
}





